src/ui/public/vislib_vis_type/vislib_renderbot.js

```diff
@@ -46,7 +46,12 @@ module.exports = function VislibRenderbotFactory(Private) {
   VislibRenderbot.prototype.buildChartData = buildChartData;
   VislibRenderbot.prototype.render = function (esResponse) {
     this.chartData = this.buildChartData(esResponse);
-    this.vislibVis.render(this.chartData, this.uiState);
+    // to allow legend to render first (wait for angular digest cycle to complete)
+    return new Promise((resolve, reject) => {
+      setTimeout(() => {
```

Comment:
This should use `$scope.$evalAsync()` so that we can be sure that angular has time to finish it's digest, and so that it runs one after we are done.